airflow/auth/managers/simple/simple_auth_manager.py

```diff
@@ -0,0 +1,166 @@
+        method: ResourceMethod,
+        details: ConfigurationDetails | None = None,
+        user: BaseUser | None = None,
+    ) -> bool:
+        return self._is_authorized(method=method, roles_to_allow=[SimpleAuthManagerRole.OP.value])
+
+    def is_authorized_connection(
+        self,
+        *,
+        method: ResourceMethod,
+        details: ConnectionDetails | None = None,
+        user: BaseUser | None = None,
+    ) -> bool:
+        return self._is_authorized(method=method, roles_to_allow=[SimpleAuthManagerRole.OP.value])
+
+    def is_authorized_dag(
+        self,
+        *,
+        method: ResourceMethod,
+        access_entity: DagAccessEntity | None = None,
+        details: DagDetails | None = None,
+        user: BaseUser | None = None,
+    ) -> bool:
+        return self._is_authorized(
+            method=method,
+            roles_to_allow=[SimpleAuthManagerRole.USER.value, SimpleAuthManagerRole.OP.value],
+        )
+
+    def is_authorized_dataset(
+        self, *, method: ResourceMethod, details: DatasetDetails | None = None, user: BaseUser | None = None
+    ) -> bool:
+        return self._is_authorized(method=method, roles_to_allow=[SimpleAuthManagerRole.OP.value])
+
+    def is_authorized_pool(
+        self, *, method: ResourceMethod, details: PoolDetails | None = None, user: BaseUser | None = None
+    ) -> bool:
+        return self._is_authorized(method=method, roles_to_allow=[SimpleAuthManagerRole.OP.value])
+
+    def is_authorized_variable(
+        self, *, method: ResourceMethod, details: VariableDetails | None = None, user: BaseUser | None = None
+    ) -> bool:
+        return self._is_authorized(method=method, roles_to_allow=[SimpleAuthManagerRole.OP.value])
+
+    def is_authorized_view(self, *, access_view: AccessView, user: BaseUser | None = None) -> bool:
+        return self._is_authorized(method="GET")
+
+    def is_authorized_custom_view(
+        self, *, method: ResourceMethod | str, resource_name: str, user: BaseUser | None = None
+    ):
+        return self.is_logged_in()
+
+    def register_views(self) -> None:
+        self.appbuilder.add_view_no_menu(
+            SimpleAuthManagerAuthenticationViews(
+                users=self.appbuilder.get_app.config.get("SIMPLE_AUTH_MANAGER_USERS", [])
+            )
+        )
+
+    def _is_authorized(
+        self,
+        *,
+        method: ResourceMethod,
+        roles_to_allow: list[str] | None = None,
+    ):
+        """
+        Return whether the user is authorized to access a given resource.
+
+        :param method: the method to perform
+        :param roles_to_allow: list of roles giving access to the resource, if the user's role is one of these roles, they have access
+        """
+        user = self.get_user()
+        if not user:
+            return False
+        role = user.get_role()
+        if role == SimpleAuthManagerRole.ADMIN.value:
+            return True
+        if method == "GET":
+            return True
```

Comment:
Hmm, don't we want to check the users roles for GETs too?

Reply:
e.g. wouldn't this let viewer see connections, for example?

Reply:
Ah.. Good point @jedcunningham -> I think it really depends on the granularity level of the basic model implemented here. I'd say we do not have to follow the exact "default" model we have in Airflow - we can simplify it (because if we don't then eventually we will have to implement all the resource management complexity that the FAB resource access model had)

I think we should be very precise about the security model that "simple" auth manager implements - and it does not have to be (or even **should not be**)  *production ready*. The assumption is that it will **only** be used in development and while it should be "secure by design" - it does not mean that it has to be same as what users had before. 

But yeah. I think it would be great if that SimpleAutManager reflects 1-1 our [Security Model](https://airflow.apache.org/docs/apache-airflow/stable/security/security_model.html) and reflects the basic roles defined there. And yes "Connection Configuration User" is described as a "role" there and "Viewer" should not have access to configuration of the connections.

Reply:
So possibly we should adapt the manager to reflect our security model 1-1 @vincbeck by default then?

Reply:
It feels simpler yet to just not even have the concept of "roles" at all. Not sure what that does for multi-team, but in isolation that's definitely what I'd advocate for.

Reply:
Oh good point! I missed that. My intention was to create a 1:1 model with our current security model but I missed the fact that "Viewer" users do not have access to config, connection, pool etc. I'll update it

Reply:
It could be this way as well - I am good both ways as soon as we have the model of that auth manager captured and explained so that we (and dev users) can know what to expect.

Reply:
> It feels simpler yet to just not even have the concept of "roles" at all. Not sure what that does for multi-team, but in isolation that's definitely what I'd advocate for.

How would you map a user to a set of permissions then?

Reply:
I updated the security model

Reply:
> How would you map a user to a set of permissions then?

I wouldn't. Just have all users have admin. I think that fits better with the dev/test use case anyways 🤷.

Reply:
I think it can be interesting for dev/test to be able to switch between roles. I can be useful to test resource access when developing Airflow

Reply:
If it was way more granular, sure! Less convinced the "all or nothing" level that the roles gives us is really that useful though.

Just a concrete example - I've relatively recently varied DAG level access between users while fixing some bugs. These roles don't offer that level of flexibility though.

Anyways, supporting roles is fine. Just my 2c that we could probably do without, and it further reinforces that this auth manager should (ideally) NOT be production ready.

Reply:
Let's see what others think :) Happy to remove these roles if this what others think as well

Reply:
Hmm. I think having 1-1 roles definition here is good. It will be easier for example to test various potential security issues related to role access etc. if we have it. So I see value in having those roles (and it's also educational value - for both users and maintainers).

BTW. I think it would be nice to generate the passwords dynamically for users and store them locally (in some cache) rather than configure them in the confiugration (and display in the output log) together with warning (!!!!THIS IS FOR DEVELOPMENT ONLY. DO NOT USE IT FOR PRODUCTION EVER!!!), Together with similar warning displayed in the UI. 

This will be secure enough and it will also make it easy to use in development environment - because the passwords will survive restarting and upgrading Airflow - but will be **certainly** not production ready.

We could even store them in breeze in a place that will survice container restarts (/files) folder - or even add a new dedicated place for such cache.

Reply:
I'm happy with that, it's a good balance. Maybe a dotfile in AIRFLOW_HOME?

Reply:
Yeah. The only issue that AIRFLOW_HOME is not "retained" across breeze runs - so if you add something there in Breeze, it will be lost when container is down. However we could add some specific volume mapping in breeze or use one of the existing ones here: https://github.com/apache/airflow/blob/main/dev/breeze/src/airflow_breeze/utils/docker_command_utils.py#L67

Reply:
> BTW. I think it would be nice to generate the passwords dynamically for users and store them locally (in some cache) rather than configure them in the confiugration (and display in the output log) together with warning (!!!!THIS IS FOR DEVELOPMENT ONLY. DO NOT USE IT FOR PRODUCTION EVER!!!), Together with similar warning displayed in the UI.

I am trying to understand the flow. When the generation would happen? Through a CLI command? I agree on the warning displayed in the UI. I can add that as follow-up PR. I dont want to pollute anymore this PR.

Reply:
See how standalone works

Reply:
I see, I actually missed the "and display in the output log" from your comment, now it makes sense :) Yep, I am happy with that, I'll update this PR

Reply:
I updated the PR and I now generate the password and save it in `generated/simple_auth_manager_passwords.json.generated`. I print out the password in the webserver console so that the user can access the password (they can also access it directly in the file `generated/simple_auth_manager_passwords.json.generated`)